src/xtgeo/cube/_cube_import.py

```diff
@@ -127,13 +157,218 @@ def import_segy(sfile: xtgeo._XTGeoFile) -> Dict:
+    idiff = np.diff(ilines_case)
+    xdiff = np.diff(xlines_case)
+    ispacing = int(np.abs(idiff[idiff != 0]).min())
+    xspacing = int(np.abs(xdiff[xdiff != 0]).min())
+
+    ncol = int(abs(ilines_case.min() - ilines_case.max()) / ispacing) + 1
+    nrow = int(abs(xlines_case.min() - xlines_case.max()) / xspacing) + 1
+    nlay = data.shape[1]
+
+    values = np.full((ncol, nrow, nlay), xtgeo.UNDEF, dtype=np.float32)
+    traceidcodes = np.full((ncol, nrow), 2, dtype=np.int64)
+
+    ilines_shifted = (ilines_case / ispacing).astype(np.int64)
+    ilines_shifted -= ilines_shifted.min()
+    xlines_shifted = (xlines_case / xspacing).astype(np.int64)
+    xlines_shifted -= xlines_shifted.min()
+
+    values[ilines_shifted, xlines_shifted, :] = data
+    values = _validate_cube_values(values)
+    traceidcodes[ilines_shifted, xlines_shifted] = traceidcodes_input
+
+    # generate new ilines, xlines vector with unique values
+    ilines = np.array(
+        range(ilines_case.min(), ilines_case.max() + ispacing, ispacing), dtype=np.int32
+    )
+    xlines = np.array(
+        range(xlines_case.min(), xlines_case.max() + xspacing, xspacing), dtype=np.int32
+    )
+
+    attrs = _geometry_incomplete_traces(
+        segyfile,
+        ncol,
+        nrow,
+        ilines,
+        xlines,
+        ilines_case,
+        xlines_case,
+        ispacing,
+        xspacing,
+    )
+
+    attrs["ncol"] = ncol
+    attrs["nrow"] = nrow
+    attrs["nlay"] = nlay
+    attrs["ilines"] = ilines
+    attrs["xlines"] = xlines
+    attrs["values"] = values
+    attrs["traceidcodes"] = traceidcodes
+    return attrs
+
+
+def _inverse_anyline_map(anylines: dict) -> Dict:
+    """Small helper function to get e.g. inline 2345: [0, 1, 2, .., 70].
+
+    I.e. to get a mapping between inline number and a list of possible indices
+
+    """
+    anyll = {}
+    for ind, key in enumerate(anylines):
+        if key not in anyll:
+            anyll[key] = list()
+        anyll[key].append(ind)
+    return anyll
+
+
+def _find_long_line(anyll: dict, nany: int) -> List:
+    """Helper function; get index of vectors indices to be used for calculations.
+
+    Look for a "sufficiently" long inline/xline, as long distance between points
+    increases accuracy.
+
+    """
+    minimumlen = int(nany / 1.2)
+    maxlenfound = -1
+
+    keepresult = []
+    for _, indices in anyll.items():
+        result = [indices[0], indices[-1]]
+        indlen = len(indices)
+        if indlen > maxlenfound:
+            maxlenfound = indlen
+            keepresult = deepcopy(result)
+
+        if indlen >= minimumlen:
+            break
+
+    if not keepresult or abs(keepresult[1] - keepresult[0]) == 0:
+        raise RuntimeError("Not able to get inline or xline vector for geometry")
+    return keepresult
+
+
+def _geometry_incomplete_traces(
+    segyfile: segyio.segy.SegyFile,
+    ncol: int,
+    nrow: int,
+    ilines: list,
```

Comment:
If you would like to inject more information in the types using `List[float]` here is possible:
```
from typing import List
```